psyneulink/core/components/functions/transferfunctions.py

```diff
@@ -1511,7 +1553,11 @@ def derivative(self, input, output=None, context=None):
         gain = self._get_current_function_param(GAIN, context)
         leak = self._get_current_function_param(LEAK, context)
 
-        return gain if input > 0 else gain * leak
+        input = np.asarray(input).copy()
```

Comment:
why does this need a copy? you're dropping the reference to original in the same statement.

Reply:
no copy is performed if the input was already a numpy array (see: https://numpy.org/doc/stable/reference/generated/numpy.asarray.html)

Reply:
so? what do you need to call `copy()`?

Reply:
wanted to prevent inplace modification of the array (incase it messed with stored state in the broader context of PsyNeuLink)

if this isn't an issue, I can remove the copy

Reply:
ah, so the situation to avoid is the input is already an np.ndarray and the lines below would modify it. Yeah, that makes sense